GVFS/GVFS.Common/OrgInfoApiClient.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+
+namespace GVFS.Common
+{
+    /// <summary>
+    /// Class that handles communication with a server that contains version information.
+    /// </summary>
+    public class OrgInfoApiClient

```

Comment:
What is the rationale for having `OrgInfoApiClient` be a different component from `QueryGVFSVersionFromOrgInfo` given that both only do one thing.. query an endpoint for a version.

Can't `OrgInfoApiClient` be the only component and have an `IOrgInfoApiClient` interface, removing the `QueryGVFSVersionFromOrgInfo` and `IQueryGVFSVersionFromOrgInfo` types?

Reply:
This is something I had been thinking about...

The reason why they are separate classes is that `OrgInfoApiClient` is more flexible, and you can use it to query the API (e.g. multiple queries for different query params) with the same instance.

`IQueryGVFSVersionFromOrgInfo` is a wrapper / adapter that is more specific to how the Upgrader wants to call it, and is a bit more restricted... The constructed instance can only query 1 set of parameters.